index.d.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Style of the box border.
+ */
+export interface BorderStyle {
+	readonly topLeft: string;
+	readonly topRight: string;
+	readonly bottomLeft: string;
+	readonly bottomRight: string;
+	readonly horizontal: string;
+	readonly vertical: string;
+}
+
+/**
+ * Spacing used for `padding` and `margin`.
+ */
+export interface Spacing {
+	readonly top: number;
+	readonly right: number;
+	readonly bottom: number;
+	readonly left: number;
+}
+
+export interface Options {
+	/**
+	 * Color of the box border.
+	 */
+	readonly borderColor?: string;
```

Comment:
Can you define the named colors supported here so it shows up in autocomplete?

Reply:
Since hex is supported, it would be difficult to supply a list of all possible colors. Note that [this issue](https://github.com/Microsoft/TypeScript/issues/12687) would prevent me from just tacking on ` | string` at the end of a union of chalk colors. I also don't want to limit TS to just chalk colors.

Reply:
[This](https://github.com/Microsoft/TypeScript/issues/6579) is also an interesting read.

Reply:
> Note that this issue would prevent me from just tacking on | string at the end of a union of chalk colors.

Have you checked if it's still a limitation? That issue is old.

I still think we should do it. And also open an new issue about it. The TS team might have improved their decision process since then (Focusing on usability over micro perf improvements).

Reply:
Opened Microsoft/TypeScript#29729. Fingers crossed.